### ISSUE TITLE
Some odd behaviour with Transformer includes

### ISSUE THREAD

**Issue by almightynassar** [thephpleague/fractal#423]
I'm having some issues where some data is included in the child resource that has the same name as an include in the parent (but oddly some cases do not do this). Expected behaviour (for me) is that only the parent will include the data, while the child will only include it if explicitly scoped.

So items A and B both include C. Item A includes B.  When I run `parseIncludes(['B', 'C'])` on resource A, I get A.C AND A.B.C (I only want A.B). I do not want this duplication of the same item.

This does not happen all the way down, as I still need to request the scoped includes for children after the first. So if items A and C both include D,  `parseIncludes(['B', 'C', 'D'])` will only give A.C, A.D, and A.B.C (again, only wanted A.B). D is declared the same in `$availableIncludes` in both transformers (as is C in the above). `parseIncludes(['B.C.D'])` will result in A.B.C.D, as expected.

I know I could give each includes it's own unique name so that it does not clash with other includes, but this would be tedious to manage long term. I could also just use parseIncludes(['B.C]) which will just give A.B.C and avoid duplicated items, which is my preferred solution for now.

Basically I'm looking to see if there is a way to avoid the first child (and only the first child, it seems) running includes that are meant only for the root.

Let me know if you need a better explanation (or proper code examples).

**Comment by willishq** [comment]
Which serializer are you using?

**Comment by almightynassar** [comment]
Thanks for the reply.

It's a custom one because I needed to emulate some old behavior. So (I'm guessing) the relevant function in this case would be (let me know if you need a full dump of the Serializer):

```
/**
 * Serialize the included data.
 *
 * @param ResourceInterface $resource
 * @param array             $data
 *
 * @return array
 */
public function includedData(ResourceInterface $resource, array $data)
{
    return $data;
}
```

I'll look into this a bit more later (and swap out serializers to see if the odd behavior occurs), but I'm assuming that the thinking is that I probably should do some checks on ResourceInterface to see if the includes field is suitable or not?

Also, as a side note, the issue only occurred for some objects and not others. And it's not a random occurrence; objects that I notice that duplicate the includes will ALWAYS duplicate the includes, and those that don't will never duplicate includes.

Although I've used a work around I'll do some deeper digging and report back in case people run into the same problem.

**Comment by greydnls** [comment]
@almightynassar I'm going to close this issue for now. Feel free to reopen it when you have more information.
